src/kernel/heap.zig

```diff
@@ -0,0 +1,388 @@
+const std = @import("std");
+const mem = std.mem;
+const testing = std.testing;
+const Allocator = std.mem.Allocator;
+const Bitmap = @import("bitmap.zig").Bitmap(usize);
+
+pub const Error = error{ /// A value provided isn't a power of two
```

Comment:
This does follow the stdlib convention of each file having its own Error struct though (such as std.mem.Allocator.Error`). Plus anyone using it would do `const heap = @import("heap.zig"); heap.Error` anyway. I think calling it HeapError would be unnecessarily verbose.

Reply:
hmm, I guess so, then can change the other Errors in the other files, this can be done as another PR?

Reply:
Sounds good to me :)